auth/service.go

```diff
@@ -0,0 +1,331 @@
+package auth
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+
+	"crypto/rsa"
+	"github.com/dgrijalva/jwt-go"
+	commonconf "github.com/fabric8-services/fabric8-common/configuration"
+	commonerrs "github.com/fabric8-services/fabric8-common/errors"
+	"github.com/fabric8-services/fabric8-common/log"
+	authclient "github.com/fabric8-services/fabric8-tenant/auth/client"
+	"github.com/fabric8-services/fabric8-tenant/configuration"
+	goaclient "github.com/goadesign/goa/client"
+	goajwt "github.com/goadesign/goa/middleware/security/jwt"
+	"github.com/pkg/errors"
+	"gopkg.in/square/go-jose.v2"
+	"net/http"
+
+	"encoding/json"
+	"github.com/fabric8-services/fabric8-wit/rest"
+	errs "github.com/pkg/errors"
+)
+
+type Service struct {
+	Config        *configuration.Data
+	ClientOptions []commonconf.HTTPClientOption
+	SaToken       string
+}
```

Comment:
is adding interface auth_service is useful, like we have for cluster_service ?

Reply:
For now, it wouldn't bring any additional value as there is no additional implementation of the `auth.Service` interface - that is also why I didn't add it there. 
In the future, I have a plan to add another dummy impl to support local testing without running auth client, then having the interface there would make sense.
As for the `cluster.Service`, yeah, I could have removed it to make the code consistent,I just didn't see it as a big deal :-). 
I can do anything of these options:
* remove interface from cluster
* add interface to auth
* keep it as it is

I don't have strong opinion on it ;-)

Reply:
I am fine with - keep it as it is :+1: